src/utils/configurations/useConfigurationsDispatchBase.js

```diff
@@ -0,0 +1,81 @@
+import { useDataEngine, useAlert } from '@dhis2/app-runtime'
+import { useCallback } from 'react'
+import { configurationsReducer } from './configurationsReducer.js'
+import { DATASTORE_ID } from './constants.js'
+
+const UPDATE_CONFIGURATIONS_MUTATION = {
+    resource: 'dataStore',
+    id: DATASTORE_ID,
+    type: 'update',
+    data: ({ newConfigurations }) => newConfigurations,
+}
+
+/**
+ * Returns a `dispatch` function, intended to a `data` parameter with the shape
+ * { type, payload }.
+ * type is an action type, which can be found in configurationsReducer.js
+ * payload is the required information to complete the relevant action
+ *
+ * Invokes the reducer function found in configurationsReducer.js to get a new
+ * configurations object, then sets the local state to that new object,
+ * attempts to sync that object with the server, and handles any errors
+ *
+ * Doesn't exactly use `useReducer` under the hood, but matches the same
+ * behavior, pattern, and function signature, so the guidelines at
+ * https://react.dev/reference/react/useReducer are relevant and helpful.
+ * (`useState` is used here to help integrate with optimistic updates and
+ * error checking)
+ * 
+ * This hook is a 'base' that can be used in useSetUpConfigurations, before
+ * the setConfigurations context is available, or for the 
+ * useConfigurationsDispatch hook intended to be supported by the 
+ * ConfigurationsProvider
+ */
+export const useConfigurationsDispatchBase = (setConfigurations) => {
```

Comment:
I refactored this out to prepare to process configurations at the useSetupConfigurations step, but as we discussed, I ended up going checking the default COC ID at edit time